src/tests/illumidesk/authenticators/test_lti11_authenticator.py

```diff
@@ -298,6 +306,63 @@ async def test_authenticator_returns_auth_state_with_empty_lis_outcome_service_u
                 'course_id': 'intro101',
                 'lms_user_id': '185d6c59731a553009ca9b59ca3a885100000',
                 'user_role': 'Learner',
+                'workspace_type': 'notebook',
+            },
+        }
+        assert result == expected
+
+
+@pytest.mark.asyncio
+@patch('illumidesk.authenticators.authenticator.LTI11LaunchValidator')
+async def test_authenticator_returns_default_workspace_type_when_missing(lti11_authenticator,):
+    """
+    Do we get the default workspace_type when its not sent with the launch request?
+    """
+    with patch.object(LTI11LaunchValidator, 'validate_launch_request', return_value=True):
+        authenticator = LTI11Authenticator()
+        args = factory_lti11_complete_launch_args('canvas', 'Learner', 'notebook')
+        args['lis_outcome_service_url'] = [b'']
```

Comment:
shouldn't this be replaced with the `workspace_type` field to make sure that we're passing a null/empty value?

Reply:
Right, wrong key replacement. I'll update it with the new auth_state_dict fixture.

Reply:
So I'm noticing that if we remove `args['lis_outcome_service_url'] = [b'']` then the control file isn't written and the test fails. Is that expected?

Reply:
I dug around a little more and that key is required. I took a not of that so we can handle that exception with LTI 1.1 in the future but we should do it in a separate PR.